src/http-gateways/trustless-gateway.md

```diff
@@ -302,31 +302,73 @@ of their presence in the DAG or the value assigned to the "dups" parameter, as
 the raw data is already present in the parent block that links to the identity
 CID.
 
-## CAR `meta` (content type parameter)
+## `meta` (content type parameter)
 
-The `meta=eof` parameter allows clients to request the server to include additional metadata about the
-CAR to be included at the end of the response body.
+The `meta` parameter allows clients to request the server to include additional metadata about the CAR along with the response body.
 
-This parameter SHOULD only be used with CAR `version=1`.
-Values other than `eof` SHOULD be ignored.
+The value of this parameter includes both the location where the metadata is given (e.g. `eof`) as well as the type of data received (e.g. `json`) separated by a `+`, to give a value such as `meta=eof+json`
 
-When the parameter is not set, the server must not add any extra CAR blocks to the response.
+When the location parameter is set to `eof`, which is currently the only supported value, the server SHOULD respond with <Response body as CARv1 stream> <0x00 byte> <Metadata>.
 
-The metadata block is a regular CAR block with the following properties:
+The only supported value for the data type parameter is `json`. This signifies that the metadata MUST be a JSON object.
 
-- CID specifies multicodec `car-metadata` (`0x04ff`), see
-  [multicodec#334](https://github.com/multiformats/multicodec/pull/334).
+This parameter MUST only be used with CAR `version=1`.
 
-- The payload contains metadata encoded as DAG-CBOR.
+When the parameter is not set or does not equal `eof+json`, the server SHOULD not add any extra blocks to the response, neither the 0x00 byte nor any metadata.
 
-The metadata MUST include the following fields:
+When `meta=eof+json`, the JSON object SHOULD conform to the following [JSON schema](https://json-schema.org/).
 
-- `len` - byte length of the CAR data (excluding the metadata block)
-- `b3h` - Blake3 hash (checksum) of the CAR data (excluding the metadata block).
-- `b3h_sig` - A signature over `<len><b3h><request>` using server's Ed2559 identity.
-  - `len` is encoded as `varint`,
-  - `b3h` is encoded as 32 bytes,
-  - The effective query as executed by the gateway. This query is the request url - path and query string arguments.
+```json
+{
+  "properties": {
+    "description": "Properties of the response"
+    "type": "object"
+  },
+  "error": {
+    "description": "Error message"
+    "type": "string"
+  },
+  "sig": {
+    "description": "A signature, using the server's Ed2559 identity, over the metadata properties object"
+    "type": "string"
+  }
+}
```

Comment:
Great idea to use JSON schema 👍🏻 

In JSON schema, all properties are optional. So with the schema above, the following objects are valid metadata blocks:
- `{}` (an empty object)
- `{properties: {}}` (no signature)
- `{error: {}}` (no signature)
- `{signature: "..."}` (signature over empty metadata)

My proposal:

```suggestion
{
  "type": "object",
  "properties": {
    "data": {
      "description": "Metadata about the request and response"
      "type": "object"
    },
    "sig": {
      "description": "A signature, using the server's Ed2559 identity, over the 'data' object. The signature data MUST be encoded using hex encoding."
      "type": "string"
    }
  }
  "required": ["props", "sig"]
}
```

You can learn more about JSON Schema here: https://json-schema.org/learn/getting-started-step-by-step

Reply:
I left them all as optional as there may be other use cases that want to use this block in a completely different way. In Rod's example, he wanted either properties or error, but didn't have a signature. I think it might be hard for us to require a signature in the json object and get buy in from Lidel

Reply:
I have added the JSON schema wrapper around the existing object but left the required array empty for now